### ISSUE TITLE
org.mobicents.diameter.dictionary.AvpDictionary method parseDictionary

### ISSUE THREAD

**Issue by NarimanAB** [RestComm/jdiameter#77]
```
public void parseDictionary(InputStream is) throws Exception {
    // we override default conf here.
    this.stackDictionary = (DictionaryImpl) DictionarySingleton.getDictionary();
    this.avpMap.clear();
    this.nameToCodeMap.clear();

    //dont like that....
    // fill AVP Map
    Map<org.jdiameter.api.validation.AvpRepresentation, org.jdiameter.api.validation.AvpRepresentation> map = this.stackDictionary.getAvpMap();
    for (org.jdiameter.api.validation.AvpRepresentation key:map.keySet()) {
      AvpRepresentationImpl value = (AvpRepresentationImpl) map.get(key);
      AvpRepresentation avp = new AvpRepresentation(value);
      this.avpMap.put(avp, avp);
      this.nameToCodeMap.put(avp.getName(), avp);
    }
  }
```

the question is why to pass "InputStream is" as a parameter and never use it...

**Comment by NarimanAB** [comment]
this my code to load custom dictionary.xml

`
AvpDictionary dictionaryFake = AvpDictionary.INSTANCE;
DictionaryImpl dictionary = (DictionaryImpl) stack.getDictionary();
dictionary.configure(this.getClass().getClassLoader().getResourceAsStream(dictionaryFile));
dictionaryFake.parseDictionary((InputStream) null);
`
It looks pretty strange especially this part - "dictionaryFake.parseDictionary((InputStream) null)". However it is the simplest way to load data from DictionaryImpl to AvpDictionary (as AvpDictionary does not actually can read file from input stream...)

**Comment by ammendonca** [comment]
Hi @NarimanAB,

Thanks for catching this! Indeed, this is not right, seems like some leftover from a refactor. We'll analyze the best way to fix this, or in case you want to give it a try, propose a fix via a Pull Request!

**Comment by ammendonca** [comment]
Fixed with PR #115.
